File: day4/part1.py
lines = [line.split(':')[1] for line in open("input.txt").read().splitlines()]

sum = 0
for line in lines:
    card_numbers, winning_numbers = [x.split() for x in line.split("|")]

    count = len([x for x in card_numbers if x in winning_numbers])

    if count > 0:
        sum += 2 ** (count - 1)

print(sum)

File: 2024/day01/src/part1.rs
use std::fs;

pub fn part1() {
    let content = match fs::read_to_string("input.txt") {
        Ok(c) => c,
        Err(error) => panic!("Problem opening the file: {error:?}"),
    };

    let lines = content.lines();

    let mut left: Vec<i64> = vec![];
    let mut right: Vec<i64> = vec![];

    for line in lines {
        let vals: Vec<i64> = line
            .split_whitespace()
            .map(|s| s.parse::<i64>().expect("Failed to parse integer"))
            .collect();

        left.push(vals[0]);
        right.push(vals[1]);
    }

    left.sort();
    right.sort();

    let mut sum = 0;
    for (l, r) in left.iter().zip(right.iter()) {
        sum += (l - r).abs();
    }

    println!("part1: {sum}");
}
